pkg/coreos/coreos_test.go

```diff
@@ -0,0 +1,61 @@
+package coreos
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2/dsl/core"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/appliance/pkg/asset/config"
+	"github.com/openshift/appliance/pkg/executer"
+	"github.com/openshift/appliance/pkg/release"
+)
+
+var _ = Describe("Test CoreOS", func() {
+	var (
+		ctrl         *gomock.Controller
+		mockExecuter *executer.MockExecuter
+		mockRelease  *release.MockRelease
+		testCoreOs   CoreOS
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		mockExecuter = executer.NewMockExecuter(ctrl)
+		mockRelease = release.NewMockRelease(ctrl)
+		coreOSConfig := CoreOSConfig{
+			Release:    mockRelease,
+			Executer:   mockExecuter,
+			EnvConfig:  &config.EnvConfig{},
+			CpuArch:    config.CpuArchitectureX86,
+			PullSecret: "'{\"auths\":{\"\":{\"auth\":\"dXNlcjpwYXNz\"}}}'",
+		}
+		testCoreOs = NewCoreOS(coreOSConfig)
+	})
+
+	It("DownloadISO - success", func() {
+		mockRelease.EXPECT().ExtractFile(machineOsImageName, fmt.Sprintf(coreOsFileName, config.CpuArchitectureX86)).Return("/path/to/file", nil).Times(1)
+		_, err := testCoreOs.DownloadISO()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("DownloadISO - fail", func() {
+		mockRelease.EXPECT().ExtractFile(machineOsImageName, fmt.Sprintf(coreOsFileName, config.CpuArchitectureX86)).Return("", errors.New("some error")).Times(1)
+		_, err := testCoreOs.DownloadISO()
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("FetchCoreOSStream - fail", func() {
```

Comment:
For testing the success flow we can have some test json file, for example: https://github.com/openshift/assisted-installer/blob/9df495e87ee5ed9c31c281c15ceee254424b56e6/src/assisted_installer_controller/assisted_installer_controller_test.go#L2026
But maybe it's a bit overkill for this simple func.